app/models/concerns/ten_digit_goods_nomenclature.rb

```diff
@@ -113,8 +113,8 @@ def code
       goods_nomenclature_item_id
     end
 
-    def short_code
-      if path_declarable?
+    def short_code(fetch_commodity_short_code: false)
```

Comment:
This feels like its repurposing an existing method when you actually need to be able to call just part of the existing method? How about breaking it up instead? Then avoids the `NullObject` overrides as well, eg

```
class GoodsNomenclature
    def some_other_short_code
      short_code
    end
  end
end

module TenDigitGoodsNomenclature
  def short_code
    ns_declarable? ? goods_nomenclature_item_id : some_other_short_code
  end

  def some_other_short_code
    case goods_nomenclature_item_id
        when /^\d+0000$/ then harmonised_system_code
        when /^\d+00$/ then combined_nomenclature_code
        else taric_code
        end
  end
end
```